packages/editor/src/context/useOnFocus.ts

```diff
@@ -0,0 +1,12 @@
+import { useEffect, useState } from 'react';
+import { useFocusWithin } from 'react-aria';
+
+export const useOnFocus = (initialValue: string, onChange: (change: string) => void) => {
+  const [isFocusWithin, setFocusWithin] = useState(false);
+  const [focusValue, setFocusValue] = useState(initialValue);
+  useEffect(() => {
+    setFocusValue(initialValue);
+  }, [initialValue]);
```

Comment:
why does this need to be in a state?
I think i should also perfectly function in a useMemo
```suggestion
  const focusValue = useMemo(() => initialValue, [initialValue]);
```